pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -1935,7 +1943,9 @@ func (ds *DistSender) sendToReplicas(
 					}
 					// Move the new lease holder to the head of the queue for the next retry.
 					if lh := routing.Leaseholder(); lh != nil {
-						transport.MoveToFront(*lh)
+						if *lh != curReplica || sameReplicaRetries < 10 {
```

Comment:
Otherwise, randomize the order?

Also pull out the magic number in a const.

Also, do we reset `sameReplicaRetries` when we try a non-leaseholder?

Add comments inline throughout.